bookwyrm/views/books/series.py

```diff
@@ -0,0 +1,60 @@
+""" books belonging to the same series """
+from django.views import View
+from django.shortcuts import get_object_or_404
+from django.template.response import TemplateResponse
+
+from bookwyrm.views.helpers import is_api_request
+from bookwyrm import models
+
+
+# pylint: disable=no-self-use
+class BookSeriesBy(View):
+    """book series by author"""
+
+    def get(self, request, author_id):
+        """lists all books in a series"""
+        series_name = request.GET.get("series_name")
+
+        if is_api_request(request):
+            pass
+
+        author = get_object_or_404(models.Author, id=author_id)
+
+        results = models.Edition.objects.filter(authors=author, series=series_name)
+
+        # when there are multiple editions of the same work, pick the closest
+        editions_of_work = results.values_list("parent_work__id", flat=True).distinct()
+
+        # filter out multiple editions of the same work
+        numbered_books = []
+        dated_books = []
+        unsortable_books = []
+        for work_id in set(editions_of_work):
+            result = (
+                results.filter(parent_work=work_id).order_by("-edition_rank").first()
+            )
+            if result.series_number:
+                numbered_books.append(result)
+            elif result.first_published_date or result.published_date:
+                dated_books.append(result)
+            else:
+                unsortable_books.append(result)
+
+        list_results = (
+            sorted(numbered_books, key=lambda book: book.series_number)
```

Comment:
This is sorting by series number _alphabetically_, so the tenth item in the series will be listed prior to the second. Coercing it to an `int` should fix it:
```python
sorted(numbered_books, key=lambda book: int(book.series_number))
```
You may need to do something similar with `book.first_published_date` below.

Reply:
Shouldn’t the published dates be okey since they are stored as `DateTimeField` so I would expect comparison to automatically work for dates?!

Reply:
I think you're right, but I didn't check it thoroughly.

Reply:
I haven't checked the code, just jumping into the conversation to say that series numbers are not always `int`s. A `float` type might be more relevant. E.g.: https://lire.boitam.eu/book/30636/s/as-yet-unsent

Also, sometimes series are not published in the right order (Star Wars, anyone?), and you get works numbered as 0.1, 0.2, etc. E.g.:  or https://www.goodreads.com/series/229503-the-locked-tomb where 1 was published in 2019 and 0.5 in 2020. Sorting by published dates won't necessarily be the same as sorting by series order.

(Apart from that it looks good, thank you for the PR :))

Reply:
Oh, good thinking re floats. Sorting by date is only a backup best-guess for when a book has a series listed but not a series number: a listed series number will take precedence.

Reply:
I actually read lots of series with `.5` ratings so no idea why I thought int would be ok haha

As hughrun said, the published date is just a backup (suggested on element) if the series number was missing.